js/ojsxc.js

```diff
@@ -392,4 +398,63 @@
             observeContactsMenu();
         }
     });
+
+    $(document).on( "click", '#jsxc_roster p', function() {
+        if (jsxc.storage.getItem('serverType') === 'internal') {
+           startInternalBackend();
+        }
+    });
+
+    function startInternalBackend() {
+       jsxc.bid = OC.currentUser.toLowerCase() + '@' + window.location.host;
+
+        jsxc.storage.setItem('jid', jsxc.bid + '/internal');
+        jsxc.storage.setItem('sid', 'internal');
+        jsxc.storage.setItem('rid', '123456');
+
+        jsxc.options.set('xmpp', {
+            url: OC.generateUrl('apps/ojsxc/http-bind')
+        });
+
+        jsxc.start(jsxc.bid + '/internal', 'internal', '123456');
+        jsxc.gui.restore();
+        jsxc.gui.roster.toggle(jsxc.CONST.SHOWN);
+        $(document).trigger('attached.jsxc');
+    }
+
+    if (jsxc.storage.getItem('serverType') === 'internal') {
+        // when the page is (re) loaded and we already know we are using the internal backend we must override
+        // the show loginBox method so that the form isn't shown when clicking the relogin link
+        jsxc.gui.showLoginBox = function () {};
+    }
+
+    $(document).on('stateChange.jsxc', function _handler(event, state) {
+        if (state === jsxc.CONST.STATE.SUSPEND) {
+            /**
+             * The first time we go into suspend mode we check if we are using the internal backend.
+             * If this is the case and the user explicitly press the "login_without_chat" button when logging
+             * into Nextcloud we know we are using another authentication mechanism (like SAML/SSO) and thus have
+             * to manually start the connection.
+             */
+            $(document).off('stateChange.jsxc', _handler);
+            if (jsxc.storage.getItem('serverType') === null) {
```

Comment:
The `serverType` is never removed, so this only works once.

Reply:
This is intentional, if the serverType is know we can directly check if its interanl here: https://github.com/nextcloud/jsxc.nextcloud/pull/67/files#diff-4f938c34e352f51c3a310a503db3e0f7R451